src/Solvers/heptadiagonal_iterative_solver.jl

```diff
@@ -293,25 +290,8 @@ function fill_boundaries_z!(coeff_d, coeff_bound_z, Az, N, ::Type{Periodic})
     end
 end
 
-function solve!(x, solver::HeptadiagonalIterativeSolver, b, Δt)
-    arch = architecture(solver.matrix)
-    
-    # update matrix and preconditioner if time step changes
-    if Δt != solver.previous_Δt
-        constructors = deepcopy(solver.matrix_constructors)
-        M = prod(solver.problem_size)
-        update_diag!(constructors, arch, M, M, solver.diagonal, Δt, 0)
-        solver.matrix = arch_sparse_matrix(arch, constructors) 
-
-        unsafe_free!(constructors)
-
-        solver.preconditioner = build_preconditioner(Val(solver.preconditioner_method),
-                                                         solver.matrix,
-                                                         solver.preconditioner_settings)
-
-        solver.previous_Δt = Δt
-    end
-    
+function solve!(x, solver::HeptadiagonalIterativeSolver, b, dt)
```

Comment:
Why dt as arg here?

Reply:
Maybe it should just be a generic `args...` like the other solvers.

Reply:
Sounds good

Reply:
But there is no extra argument needed anyway. 
is it only related to function’s signature?

Reply:
Ok yeah I don't understand. `dt` isn't used.